test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,459 @@
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = writerList.stream().map(writer ->
+                    startWriting(writer)).collect(Collectors.toList());
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = readerList.stream().map(reader ->
+                    startReading(reader))
+                    .collect(Collectors.toList());
+
+            //increase the number of writers to trigger scale
+            List<EventStreamWriter<Long>> writerList1 = new ArrayList<>(ADD_NUM_WRITERS);
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < ADD_NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer1 = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                writerList1.add(writer1);
+            }
+
+            //start writing asynchronoulsy with newly created writers
+            List<CompletableFuture<Void>> writerFutureList1 = writerList1.stream().map(writer ->
+                    startWriting(writer)).collect(Collectors.toList());
+
+            //run the failover test while scaling
+            performFailoverTest();
+
+            //wait for scaling
+            waitForScaling();
+
+            //bring the instances back to 3 before performing failover
+            controllerInstance.scaleService(3, true);
+            segmentStoreInstance.scaleService(3, true);
+            Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
+
+            //run the failover test after scaling
+            performFailoverTest();
+
+            log.info("Stop write flag status {}", testState.stopWriteFlag);
+            testState.stopWriteFlag.set(true);
+
+            log.info("Wait for writers execution to complete");
+            FutureHelpers.allOf(writerFutureList).get();
+
+            FutureHelpers.allOf(writerFutureList1).get();
+
+            log.info("Stop read flag status {}", testState.stopReadFlag);
+            testState.stopReadFlag.set(true);
+
+            log.info("Wait for readers execution to complete");
+            FutureHelpers.allOf(readerFutureList).get();
+
+            log.info("All writers have stopped. Setting stopReadFlag. Event Written Count:{}, Event Read " +
+                    "Count: {}", testState.eventWriteCount.get(), testState.eventsReadFromPravega.size());
+            assertEquals(testState.eventWriteCount.get(), testState.eventsReadFromPravega.size());
+            assertEquals(testState.eventWriteCount.get(), new TreeSet<>(testState.eventsReadFromPravega).size()); //check unique events.
+
+            log.info("Closing writers");
+            writerList.forEach(writer -> writer.close());
```

Comment:
I don't see the writers in writerList1 getting closed.
Also maybe there could be a better name for that variable.